.github/workflows/dev-container.yml

```diff
@@ -0,0 +1,28 @@
+name: 'devcontainer CI' 
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build_devcontainer:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout from github
+        uses: actions/checkout@v3
+      - name: extract the github reference
+        run: echo "BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
+      - name: "image name from repo name"
+        id: docker_image_name
+        run: echo "docker_image=${{ github.repository }}" | tr '[:upper:]' '[:lower:]' |sed 's/[^0-9,a-z,A-Z,=,_,\/]/-/g' >>${GITHUB_OUTPUT}
+      - name: Build and run dev container task
+        uses: devcontainers/ci@v0.3
+        with:
+          imageName: lcas.lincoln.ac.uk/devcontainer/${{ steps.docker_image_name.outputs.docker_image }}
+          push: never
+          imageTag: ${{ env.BRANCH }}
+          #runCmd: "bash .devcontainer/run-ci.sh"
```

Comment:
This is a new CI workflow, that ensures that the devcontainer image can be build. Basically it guarantees that the devcontainer image on the main branch always works